homeassistant/components/template/select.py

```diff
@@ -187,34 +206,38 @@ def __init__(
         config: dict,
     ) -> None:
         """Initialize the entity."""
-        super().__init__(hass, coordinator, config)
+        TriggerEntity.__init__(self, hass, coordinator, config)
+        AbstractTemplateSelect.__init__(self, config)
+
+        if CONF_STATE in config:
+            self._to_render_simple.append(CONF_STATE)
+
+        self._attr_name = name = self._rendered.get(CONF_NAME, DEFAULT_NAME)
+
         # Scripts can be an empty list, therefore we need to check for None
         if (select_option := config.get(CONF_SELECT_OPTION)) is not None:
-            self.add_script(
-                CONF_SELECT_OPTION,
-                select_option,
-                self._rendered.get(CONF_NAME, DEFAULT_NAME),
-                DOMAIN,
-            )
+            self.add_script(CONF_SELECT_OPTION, select_option, name, DOMAIN)
 
-    @property
-    def current_option(self) -> str | None:
-        """Return the currently selected option."""
-        return self._rendered.get(CONF_STATE)
+    def _handle_coordinator_update(self):
+        """Handle updated data from the coordinator."""
+        self._process_data()
 
-    @property
-    def options(self) -> list[str]:
-        """Return the list of available options."""
-        return self._rendered.get(ATTR_OPTIONS, [])
-
-    async def async_select_option(self, option: str) -> None:
-        """Change the selected option."""
-        if self._config[CONF_OPTIMISTIC]:
-            self._attr_current_option = option
+        if not self.available:
+            self.async_write_ha_state()
+            return
+
+        write_ha_state = False
+        for key, attr, validator in (
+            (ATTR_OPTIONS, "_attr_options", vol.All(cv.ensure_list, [cv.string])),
+            (CONF_STATE, "_attr_current_option", cv.string),
+        ):
+            if (rendered := self._rendered.get(key)) is not None:
+                setattr(self, attr, validator(rendered))
+
+        if len(self._rendered) > 0:
+            # In case any non optimistic template
+            write_ha_state = True
+
+        if write_ha_state:
+            self.async_set_context(self.coordinator.data["context"])
```

Comment:
```suggestion
        if len(self._rendered) > 0:
            # In case any non optimistic template
            self.async_set_context(self.coordinator.data["context"])
```